internal/auth/oidc.go

```diff
@@ -0,0 +1,251 @@
+	}
+
+	if cfg.OIDCClientID == "" {
+		log.Error().Msg("OIDC client ID is empty")
+		return nil, fmt.Errorf("OIDC client ID is required")
+	}
+
+	if cfg.OIDCClientSecret == "" {
+		log.Error().Msg("OIDC client secret is empty")
+		return nil, fmt.Errorf("OIDC client secret is required")
+	}
+
+	if cfg.OIDCRedirectURL == "" {
+		log.Error().Msg("OIDC redirect URL is empty")
+		return nil, fmt.Errorf("OIDC redirect URL is required")
+	}
+
+	scopes := []string{"openid", "profile", "email"}
+
+	ctx := context.Background()
+	provider, err := oidc.NewProvider(ctx, cfg.OIDCIssuer)
+	if err != nil {
+		log.Error().Err(err).Msg("failed to initialize OIDC provider")
+		return nil, fmt.Errorf("failed to initialize OIDC provider: %w", err)
+	}
+
+	oidcConfig := &oidc.Config{
+		ClientID: cfg.OIDCClientID,
+	}
+
+	handler := &OIDCHandler{
+		log: log,
+		config: &OIDCConfig{
+			Enabled:      cfg.OIDCEnabled,
+			Issuer:       cfg.OIDCIssuer,
+			ClientID:     cfg.OIDCClientID,
+			ClientSecret: cfg.OIDCClientSecret,
+			RedirectURL:  cfg.OIDCRedirectURL,
+			Scopes:       scopes,
+		},
+		provider: provider,
+		verifier: provider.Verifier(oidcConfig),
+		oauthConfig: &oauth2.Config{
+			ClientID:     cfg.OIDCClientID,
+			ClientSecret: cfg.OIDCClientSecret,
+			RedirectURL:  cfg.OIDCRedirectURL,
+			Endpoint:     provider.Endpoint(),
+			Scopes:       scopes,
+		},
+	}
+
+	log.Debug().Msg("OIDC handler initialized successfully")
+	return handler, nil
+}
+
+func (h *OIDCHandler) GetConfig() *OIDCConfig {
+	if h == nil {
+		return &OIDCConfig{
+			Enabled: false,
+		}
+	}
+	h.log.Debug().
+		Bool("enabled", h.config.Enabled).
+		Str("issuer", h.config.Issuer).
+		Msg("returning OIDC config")
+	return h.config
+}
+
+func (h *OIDCHandler) HandleLogin(w http.ResponseWriter, r *http.Request) {
+	state := generateRandomState()
+
+	h.SetStateCookie(w, r, state)
+
+	authURL := h.oauthConfig.AuthCodeURL(state)
+	http.Redirect(w, r, authURL, http.StatusFound)
+}
+
+func (h *OIDCHandler) HandleCallback(w http.ResponseWriter, r *http.Request) (string, error) {
+	h.log.Debug().Msg("handling OIDC callback")
+
+	state, err := r.Cookie("state")
+	if err != nil {
+		h.log.Error().Err(err).Msg("state cookie not found")
+		return "", fmt.Errorf("state cookie not found")
+	}
+	if r.URL.Query().Get("state") != state.Value {
+		h.log.Error().
+			Str("expected", state.Value).
+			Str("got", r.URL.Query().Get("state")).
+			Msg("state did not match")
+		return "", fmt.Errorf("state did not match")
+	}
+
+	oauth2Token, err := h.oauthConfig.Exchange(r.Context(), r.URL.Query().Get("code"))
+	if err != nil {
+		h.log.Error().Err(err).Msg("failed to exchange token")
+		return "", fmt.Errorf("failed to exchange token: %w", err)
+	}
+
+	rawIDToken, ok := oauth2Token.Extra("id_token").(string)
+	if !ok {
+		h.log.Error().Msg("no id_token found in oauth2 token")
+		return "", fmt.Errorf("no id_token found in oauth2 token")
+	}
+
+	idToken, err := h.verifier.Verify(r.Context(), rawIDToken)
```

Comment:
Since autobrr will store the `client_secret` in the config or by env vars on the server side, [PKCE](https://pkg.go.dev/golang.org/x/oauth2#VerifierOption) wouldn't provide meaningful additional security. PKCE is useful when client secrets can't be stored, like in pure frontend applications where secrets would be exposed.